torch/onnx/operators.py

```diff
@@ -0,0 +1,24 @@
+import torch
+from torch.autograd import Variable
+import torch.onnx
+
+# This file contains pure pytorch operators that should eventually
+# find a home outside of the onnx/ directory
+
+
+def _size_as_variable(g, input):
+    return g.op('Shape', input)
+
+
+@torch.onnx.symbolic_override(_size_as_variable)
+def size_as_variable(var):
+    return Variable(torch.LongTensor(tuple(var.size())))
+
+
+def _view_from_variable_shape(g, input, shape):
+    return g.op('Reshape', input, shape)
+
+
+@torch.onnx.symbolic_override(_view_from_variable_shape)
+def view_from_variable_shape(var, size_var):
+    return var.view(size_var.tolist())
```

Comment:
`reshape` (instead of `view`) will also work on non-contiguous tensors (but may make copy in some cases). Not sure about the semantics you want.

Reply:
hmm I guess I'll do reshape